raiden/transfer/architecture.py

```diff
@@ -238,28 +237,35 @@ class StateManager(Generic[ST]):
 
     def __init__(
         self,
-        state_transition: Callable[[Optional[ST], StateChange], TransitionResult[ST]],
-        current_state: Optional[ST],
+        state_transition: Callable[[ST, StateChange], TransitionResult[ST]],
+        current_state: ST,
+        unapplied_state_changes: List[StateChange],
```

Comment:
Do we really want this as an extra parameter? Wouldn't it be clearer to do
```python
        for state_change in unapplied_state_changes:
            self.dispatch(state_change)
```
at the caller?

Reply:
I already have a follow-up in the works that removes the `StateManager` and fixes that.